modules/service/data-service.js

```diff
@@ -2,27 +2,56 @@ const { v1: uuidv1 } = require('uuid');
 const constants = require('../constants');
 const GraphDB = require('../../external/graphdb-service');
 
+// TODO: Discussion on this values
+const MAX_RETRIES = 10;
+const RETRY_FREQUENCY = 10; // seconds
+
 class DataService {
     constructor(ctx) {
         this.config = ctx.config;
         this.logger = ctx.logger;
         this.constants = ctx.constants;
         this.validationService = ctx.validationService;
+        this.networkService = ctx.networkService;
     }
 
     getName() {
         return this.implementation.getName();
     }
 
-    initialize() {
+    async initialize() {
         this.implementation = new GraphDB({
             repositoryName: this.config.graphDatabase.name,
             username: this.config.graphDatabase.username,
             password: this.config.graphDatabase.password,
         });
+
+        let ready = await this.healthCheck();
+        let retries = 0;
+        while (!ready && retries < MAX_RETRIES) {
+            retries += 1;
+            this.logger.warn(`Cannot connect to Triple store (${this.getName()}), retry number: ${retries}/${MAX_RETRIES}. Retrying in ${RETRY_FREQUENCY} seconds again.`);
+            await new Promise((resolve) => setTimeout(resolve, RETRY_FREQUENCY * 1000));
+            ready = await this.healthCheck();
+        }
+        if (retries === MAX_RETRIES) {
+            console.error('Triple store (GraphDB) not available, max retries reached. Stopping the node...');
```

Comment:
Use logger

Reply:
There is process.exit after this log message, so the process stops before the logger manages to log the message. 
 This was my first solution, alternatively I can insert logger.error and waiting N seconds before  shutting down the node. What do you think, which approach is better?